lang/java/reef-io/src/main/java/org/apache/reef/io/data/loading/impl/NumberedSplit.java

```diff
@@ -62,6 +62,24 @@ public String toString() {
   }
 
   @Override
+  public boolean equals(final Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    NumberedSplit<?> that = (NumberedSplit<?>) o;
+    return index == that.index;
```

Comment:
Hi, @tcNickolas . Is it enough to compare only `index`? `NumberedSplit` has two more members.
```
new NumberedSplit<>(split, splitNum, partitions[splitNum]);
```